src/jit/lower.cpp

```diff
@@ -432,17 +432,34 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
             // relink them into the proper order, i.e. loOp1 is followed by loOp2, and then
             // the loResult node.
             // (This rethreading, and that below, are where we need to address the reverse ops case).
-            loOp1->gtNext = loOp2;
-            loOp2->gtPrev = loOp1;
-            loOp2->gtNext = loResult;
-            loResult->gtPrev = loOp2;
-
-            // We will now create a new tree for the hiResult, and then thread these nodes as above.
+            // The current order is (after snipping op1 and op2):
+            // loOp1First ... loOp1->hiOp1First ... hiOp1->loOp2First ... loOp2->hiOp2First ... hiOp2
+            // The order we want is:
+            // loOp1First ... loOp1->loOp2First ... loOp2->loResult->hiOp1First ... hiOp1->hiOp2First ... hiOp2->hiResult
+            // (i.e. we swap hiOp1 and loOp2)
+
+            GenTree* loOp2First = hiOp1->gtNext;
+            GenTree* hiOp1First = loOp1->gtNext;
```

Comment:
Ah, it's not actually needed as it turns out, because the code below uses the more generic `fgInsertTreeInListAfter()` to insert the hiResult tree after the loResult. So that can be removed.